src/components/CheckoutBilling/index.tsx

```diff
@@ -1,135 +1,163 @@
+import {
+  CheckoutContext,
+  CheckoutContextInterface
+} from "../CheckoutApp/context";
 import {
   checkoutBaseUrl,
   checkoutPaymentUrl,
   checkoutShippingOptionsUrl
 } from "../CheckoutApp/routes";
+import {
+  Checkout,
+  Checkout_billingAddress
+} from "../CheckoutApp/types/Checkout";
 import { ShopContext } from "../ShopProvider/context";
-import { UPDATE_CHECKOUT_BILLING_ADDRESS } from "./queries";
+import { getShop_shop } from "../ShopProvider/types/getShop";
+import { TypedUpdateCheckoutBillingAddressMutation } from "./queries";
+import { updateCheckoutBillingAddress } from "./types/updateCheckoutBillingAddress";
 
-class CheckoutBilling extends React.Component<RouteComponentProps<{ id }>, {}> {
-  render() {
-    return (
-      <div>
-        <CheckoutContext.Consumer>
-          {({ checkout, updateCheckout, shippingAsBilling }) => {
-            const address =
-              !checkout.billingAddress && shippingAsBilling
-                ? checkout.shippingAddress
-                : checkout.billingAddress;
-            return (
-              <>
-                <Link to={checkoutBaseUrl(checkout.token)}>
-                  <div className="checkout__step checkout__step--inactive">
-                    <span>1</span>
-                    <h4 className="checkout__header">Shipping Address</h4>
-                  </div>
-                </Link>
-                <div className="checkout__content">
-                  <AddressSummary
-                    address={checkout.shippingAddress}
-                    email={checkout.email}
-                  />
-                </div>
-                <Link to={checkoutShippingOptionsUrl(checkout.token)}>
-                  <div className="checkout__step checkout__step--inactive">
-                    <span>2</span>
-                    <h4 className="checkout__header">Shipping Method</h4>
-                  </div>
-                </Link>
+const proceedToPayment = (
+  data: updateCheckoutBillingAddress,
+  checkoutCtx: CheckoutContextInterface,
+  history: H.History
+) => {
+  const canProceed = maybe(
+    () => !data.checkoutBillingAddressUpdate.errors.length
+  );
+
+  if (canProceed) {
+    checkoutCtx.updateCheckout({
+      checkout: data.checkoutBillingAddressUpdate.checkout
+    });
+    history.push(checkoutPaymentUrl(checkoutCtx.checkout.token));
+  }
+};
+
+const extractBillingData = (
+  address: Checkout_billingAddress,
+  shop: getShop_shop
+) => {
+  const hasAddress = maybe(() => !!address.country);
+  if (hasAddress) {
+    return address;
+  }
+
+  const { geolocalization, defaultCountry } = shop;
+  return {
+    ...address,
+    country: {
+      code: geolocalization.country
+        ? geolocalization.country.code
+        : defaultCountry.code,
+      country: geolocalization.country
+        ? geolocalization.country.country
+        : defaultCountry.country
+    }
+  };
+};
+
+const computeMutationVariables = (data: any, checkout: Checkout) => ({
```

Comment:
`data: any`... Can we use *a bit* stricter type?

Reply:
I'll see what I can do, this comes from the form.